pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/AbstractJsonPostOrPutServlet.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.helper.TransformToPojo;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Abstract servlet class to provide common behaviour that accepts a json payload in the body of a POST or PUT
+ * web request, parses said payload into a java class, and then operates on the object. All the json
+ * deserialization logic is encapsulated in this class.
+ *
+ * Implementations should override the processPost of processPut methods (both are allowed) to implement
+ * specific logic for either case.
+ *
+ * @author Carlos Munoz
+ */
+public class AbstractJsonPostOrPutServlet<T> extends SlingAllMethodsServlet {
+
+    private final Class<T> jsonType;
+
+    public AbstractJsonPostOrPutServlet(Class<T> jsonType) {
+        this.jsonType = jsonType;
+    }
+
+    @Override
+    protected final void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response)
+            throws ServletException, IOException {
+        TransformToPojo transformToPojo = new TransformToPojo();
+        T jsonPayload = transformToPojo.fromJson(jsonType, request.getReader());
+
+        try {
+            processPost(request, response, jsonPayload);
+        } catch (Exception e) {
+            throw new ServletException(e);
```

Comment:
Logging the exception here would help in tracing any exception in splunk.

Reply:
Do you mean logging the exception instead of throwing it? or doing both?

Reply:
Doing both. When client recieves the exception, having a log of the same on server side helps a lot in case debugging is required  :)